### ISSUE TITLE
Live streaming cannot be viewed on an iOS mobile browser

### ISSUE THREAD

**Issue by sayidhe** [Ruud14/SecurityCamera#12]
Try the address `http://<local_pi_ip>:8000/index.html` on iOS mobile browser both **Chrome** and **Safari**, only the play button on it, can not fetch the live streaming. But the link works fine on desktop browser.

**Comment by mohan51** [comment]
hi @sayidhe , is the detector functionality is working?

**Comment by mohan51** [comment]
@sayidhe because whenever i am executing the main.py it is not detecting anything

**Comment by sayidhe** [comment]
Hi @mohan51 , it works fine on my machine:

- Raspberry Pi 4B
- RS/E14 Camera V2
- Raspbian OS Lite

**Comment by mohan51** [comment]
@sayidhe we are also working on the same board but we are unable to detect ? have you done any changes in main.py or detector code?

**Comment by mohan51** [comment]
and also in django frame work which path are you giving at the time of camera addition? is it http://<pi_address>:8000/index.html or http://<pi_address>:8000/stream.mpeg?

**Comment by mohan51** [comment]
and i am using arducam and raspberrypi 4 and raspbian os

**Comment by sayidhe** [comment]
@mohan51 Which `ffmpeg` path are you use? 

When I use `sudo apt install ffmpeg`, the `ffmpeg_path` setting in **config.json** line is:

- "ffmpeg_path": "/usr/bin/ffmpeg",

If you follow the step to install and make the ffmpeg as [Install ffmpeg with h.264 support](http://jollejolles.com/installing-ffmpeg-with-h264-support-on-raspberry-pi/), then just keep the same in config.json as:

- "ffmpeg_path": "/usr/local/bin/ffmpeg"

**Comment by mohan51** [comment]
@sayidhe I used sudo apt install ffmpeg only and set the path to "ffmpeg_path": "/usr/bin/ffmpeg", in config.json and also in recorder.py

**Comment by mohan51** [comment]
> and also in django frame work which path are you giving at the time of camera addition? is it http://<pi_address>:8000/index.html or http://<pi_address>:8000/stream.mpeg?

@sayidhe

**Comment by sayidhe** [comment]
In django frame work, I use `http://<pi_address>:8000/index.html` in the camera url

**Comment by mohan51** [comment]
what type of detections it is detecting? and also is it also sending recordings???

@sayidhe

**Comment by sayidhe** [comment]
Can you see the live streaming, but can't detect anything? @mohan51

**Comment by mohan51** [comment]
> Can you see the live streaming, but can't detect anything? @mohan51

yes.and also the recordings i am not able to get.

@sayidhe

**Comment by sayidhe** [comment]
One thing, I use latest version of python and remove the packages version number in `requirements.txt`, like below

```
chardet
numpy
picamera
python-apt
requests
six
ssh-import-id
tornado
urllib3
```

Others just keep the same.

**Comment by mohan51** [comment]
ok
will try and get back to you

**Comment by sayidhe** [comment]
Can you view the live streaming on the mobile browser? @mohan51

**Comment by mohan51** [comment]
> Can you view the live streaming on the mobile browser? @mohan51

yes.. when i connected to same network

**Comment by mohan51** [comment]
what actually it detects? it will detects everything?i am every movement?

**Comment by mohan51** [comment]
@sayidhe for python-apt installation i am getting error?

**Comment by sayidhe** [comment]
There are some lines in the `config.json` which controls the detector

```
 "detector_motion_threshold": 20,
 "record_seconds_before_motion": 5,
 "record_seconds_after_motion": 12,
 "max_recording_seconds": 600,
```

- `detector_motion_threshold`: Threshold for detecting motion. The higher this number, the less motion is detected.
- `record_seconds_before_motion`: The amount of seconds that will be recorded before motion is detected.
- `record_seconds_after_motion`: The amount of seconds that will be recorded after motion. 
- `max_recording_seconds`: The maximum duration of a recording in seconds.

**Comment by mohan51** [comment]
> There are some lines in the `config.json` which controls the detector
> 
> ```
>  "detector_motion_threshold": 20,
>  "record_seconds_before_motion": 5,
>  "record_seconds_after_motion": 12,
>  "max_recording_seconds": 600,
> ```
> 
> * `detector_motion_threshold`: Threshold for detecting motion. The higher this number, the less motion is detected.
> * `record_seconds_before_motion`: The amount of seconds that will be recorded before motion is detected.
> * `record_seconds_after_motion`: The amount of seconds that will be recorded after motion.
> * `max_recording_seconds`: The maximum duration of a recording in seconds.

yeah i saw the lines and adjusted to my requirement but still unable to record or detect the motion

**Comment by sayidhe** [comment]
Do you have `read, write and execute permission` of the folder?

**Comment by mohan51** [comment]
i cloned in to my pc and working on the changes
..but for testing i used the same config file
![MicrosoftTeams-image](https://github.com/Ruud14/SecurityCamera/assets/16484606/3900db26-1806-4261-9f55-e323a3753c19)

**Comment by mohan51** [comment]
it started streaming but unable to detect the movements

@sayidhe

**Comment by sayidhe** [comment]
@mohan51  Below is my output

```
Motion detector started successfully!
Streamer started on http://192.168.*.*:8000
Started recording /home/pi/SecurityCamera/./temp_recordings/16-54-29
ffmpeg version git-2023-08-08-9583a2c Copyright (c) 2000-2023 the FFmpeg developers
...
...
...
```

**Comment by mohan51** [comment]
ffmpeg version git-2023-08-08-9583a2c Copyright (c) 2000-2023 the FFmpeg developers

i am not getting this line...do you have any idea?

@sayidhe

**Comment by mohan51** [comment]
I downloaded ffmpeg using this steps https://phoenixnap.com/kb/install-ffmpeg-ubuntu

**Comment by sayidhe** [comment]
I followed this one [Install ffmpeg with h.264 support](http://jollejolles.com/installing-ffmpeg-with-h264-support-on-raspberry-pi/) and keep `ffmpeg_path` like:

- "ffmpeg_path": "/usr/local/bin/ffmpeg"

**Comment by mohan51** [comment]
> what actually it detects? it will detects everything?i am every movement?

detects every movement? with default config settings

**Comment by mohan51** [comment]
i am done with installation.then how can i know the path of ffmpeg?how to known whether it is working or not?
when i execute "which ffmpeg". it is not showing anything


@sayidhe

**Comment by sayidhe** [comment]
Normally it is in "/usr/local/bin/ffmpeg", when you make and install. You can check with 

`whereis ffmpeg`

**Comment by mohan51** [comment]
Started recording /home/pi/SecurityCamera/./temp_recordings/16-54-29
ffmpeg version git-2023-08-08-9583a2c Copyright (c) 2000-2023 the FFmpeg developers

these two lines are not coming for me.i tried the way you installed ffmpeg.and it installed successfully but the above two lines are not coming

**Comment by sayidhe** [comment]
> Do you have `read, write and execute permission` of the folder?

@mohan51 Not sure if the above issue, you can try to changing the ownership of a folder to pi user, and run the script again.

```
sudo chown pi:pi -R /home/pi/SecurityCamera
```

Otherwise, try to re-install the OS, then follow the [installation-guide](https://github.com/Ruud14/SecurityCamera#installation-guide)

**Comment by Ruud14** [comment]
Hi @sayidhe,
Thanks for reporting.

I tried viewing the stream on safari iOS and I indeed cannot see the camera stream.

I'll mark this as a bug and look more into it.
